source/features/batch-mark-files-as-viewed.tsx

```diff
@@ -18,23 +18,28 @@ function isChecked(file: HTMLElement): boolean {
 }
 
 function batchToggle(event: delegate.Event<MouseEvent, HTMLFormElement>): void {
-	if (!event.shiftKey || !previousFile) {
+	if (!event.shiftKey) {
+		return;
+	}
+
+	previousFile ??= select('.js-file'); // #5484
+	if (!previousFile?.isConnected) {
 		return;
 	}
 
 	event.preventDefault();
 	event.stopImmediatePropagation();
 
-	const previousFileState = isChecked(previousFile);
 	const thisFile = event.delegateTarget.closest('.js-file')!;
 	const files = select.all('.js-file');
+
 	const selectedFiles = files.slice(...[
-		files.indexOf(previousFile) + 1,
+		files.indexOf(previousFile),
 		files.indexOf(thisFile) + 1,
 	].sort((a, b) => a - b));
 
 	for (const file of selectedFiles) {
-		if (isChecked(file) !== previousFileState) {
+		if (!isChecked(file)) {
```

Comment:
This is _not_ equivalent. The feature should work the same whether checking or unchecking multiple files.

Reply:
> This is _not_ equivalent. The feature should work the same whether checking or unchecking multiple files.

Thanks for pointing this out!
I just updated the codes.